wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java

```diff
@@ -175,20 +222,101 @@ public static int sizeOf(List<ValueType> args) {
         return total;
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(opcode, operand);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof ValueType)) {
+            return false;
+        }
+        ValueType that = (ValueType) other;
+        return this.opcode == that.opcode && this.operand == that.operand;
+    }
+
+    public String toString() {
+        switch (opcode) {
+            case Ref:
+            case RefNull:
+                return opcode.name() + "[" + operand + "]";
+            default:
+                return opcode.name();
+        }
+    }
+
+    /**
+     * a string representation of [ValueType] that follows JVM's naming conventions
+     */
+    public String name() {
+        return opcode.name();
+    }
+
     /**
-     * A separate holder class for ID constants.
-     * This is necessary because enum constants are initialized before normal fields, so any reference to an ID constant
-     * in the same class would be considered an invalid forward reference.
+     * basically ValueTypeOpcode, but removes FuncRef and ExternRef, which alias Ref/RefNull
      */
-    static final class ID {
-        private ID() {}
-
-        static final int ExternRef = 0x6f;
-        static final int FuncRef = 0x70;
-        static final int V128 = 0x7b;
-        static final int F64 = 0x7c;
-        static final int F32 = 0x7d;
-        static final int I64 = 0x7e;
-        static final int I32 = 0x7f;
+    public enum Variant {
```

Comment:
I'm a little confused by this abstraction:

```java
ValueType(ValueTypeOpCode, int operand);
Variant(id) == Variant(ValueTypeOpCode);
```

do we need `Variant` at all? Seems to encode a concept already included in `ValueType` or am I missing something?